### PR TITLE
Ambar chat timestamps and ui

### PR DESCRIPTION
**Pull request by eddieberklee**

### CODE REVIEW

File: app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java [hunk 1]
```diff
@@ -1,68 +1,69 @@
...
 
   public void setIsSender(boolean isSender) {
-    int color;
+    int bg_color;
+    int text_color;
     if (isSender) {
-      color = ContextCompat.getColor(rootView.getContext(), R.color.group_chat_background_color);
-      leftArrow.setVisibility(View.GONE);
-      rightArrow.setVisibility(View.VISIBLE);
+      bg_color = ContextCompat.getColor(rootView.getContext(), R.color.chat_sender);
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java]: #factor `rootView.getContext()` --> `Context context = rootView.getContext()` and use that variable instead. Little cleaner

File: app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java [hunk 2]
```diff
@@ -1,68 +1,69 @@
...
+      text_color = ContextCompat.getColor(rootView.getContext(), R.color.chat_sender_text);
       messageContainer.setGravity(Gravity.RIGHT);
+      userName.setVisibility(View.GONE);
     } else {
-      color = ContextCompat.getColor(rootView.getContext(), R.color.user_chat_background_color);
-      leftArrow.setVisibility(View.VISIBLE);
-      rightArrow.setVisibility(View.GONE);
+      bg_color = ContextCompat.getColor(rootView.getContext(), R.color.chat_sendee);
+      text_color = ContextCompat.getColor(rootView.getContext(), R.color.chat_sendee_text);
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java]: Can refactor this .getColor() call

File: app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java [hunk 3]
```diff
@@ -1,68 +1,69 @@
...
-    ((GradientDrawable) messageBox.getBackground()).setColor(color);
-    ((RotateDrawable) leftArrow.getBackground()).getDrawable()
-        .setColorFilter(color, PorterDuff.Mode.SRC);
-    ((RotateDrawable) rightArrow.getBackground()).getDrawable()
-        .setColorFilter(color, PorterDuff.Mode.SRC);
+    ((GradientDrawable) messageBox.getBackground()).setColor(bg_color);
+    messageText.setTextColor(text_color);
+  }
+
+  public void setTimestamp(Date date){
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java]: setTimestampText might be a little more clearer

File: app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java [hunk 4]
```diff
@@ -1,68 +1,69 @@
...
+    int hours = date.getHours();
+    int minutes = date.getMinutes();
+    timestampText.setText(hours + ":" + minutes);
   }
 
   public void setName(String name) {
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java:65]: I think these might be better as not being methods - they're not used more than once

File: app/src/main/java/com/compscieddy/meetinthemiddle/model/Chat.java [hunk 5]
```diff
@@ -63,7 +67,7 @@ public void update() {
     result.put("key", groupKey);
     result.put("userKey", userKey);
     result.put("chatMessage", chatMessage);
-    result.put("timestamp", timestamp);
+    result.put("currentUTCTime", currentUTCTime);
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/model/Chat.java:70]: Nice, I once forgot about updating this method once and was debugging for like 30 minutes 😒

File: app/src/main/java/com/compscieddy/meetinthemiddle/util/Util.java [hunk 6]
```diff
@@ -63,4 +68,15 @@ public static void rotateLocationInactive(View view) {
         .start();
   }
 
+  public static Date getCurrentUTCTime(){
+    // Set device's time zone to UTC
+    TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+    Calendar calendar = Calendar.getInstance();
+    // Reset device's time zone to device local time zone
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/util/Util.java]: This comment seems unrelated to the code...where is it begin reset? (maybe I just don't see it)

File: app/src/main/java/com/compscieddy/meetinthemiddle/util/Util.java [hunk 7]
```diff
@@ -63,4 +68,15 @@ public static void rotateLocationInactive(View view) {
...
+    return calendar.getTime();
+  }
+
+  public static Date getLocalTimeFromUTC(Date utcDate){
+   return new Date(utcDate.getTime() + TimeZone.getDefault().getOffset(utcDate.getTime()));
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/util/Util.java:80]: Can you add a 1-line comment explaining what you're doing. It's like almost simple enough to understand, but timezones can be a confusing concept 😃

File: app/src/main/res/drawable/chat_message_background.xml [hunk 8]
```diff
@@ -1,5 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <shape xmlns:android="http://schemas.android.com/apk/res/android" android:shape="rectangle">
-    <corners android:radius="4dp"/>
-    <solid android:color="@color/material_grey_300" />
+    <corners android:radius="15dp"/>
```
Comment (eddieberklee) [inline app/src/main/res/drawable/chat_message_background.xml]: I think moving this 15dp out to a dimens could be useful for later

File: app/src/main/res/layout/item_chat.xml [hunk 9]
```diff
@@ -1,26 +1,35 @@
...
+        android:textSize="12sp"
+        tools:text="Time Stamp"/>
+
+    <TextView
+        android:id="@+id/name_text"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textColor="@color/user_name"
+        android:textSize="12sp"
+        tools:text="Some User"/>
```
Comment (eddieberklee) [inline app/src/main/res/layout/item_chat.xml:27]: Thanks for using tools:text and not android:text 👍

File: app/src/main/res/values/colors.xml [hunk 10]
```diff
@@ -6,8 +6,11 @@
     <color name="colorPrimaryDark">#303F9F</color>
     <color name="colorAccent">@color/mitm_yellow</color>
 
-    <color name="chat_sendee">#E7E7E7</color>
-    <color name="chat_sender">#087FF6</color>
+    <color name="chat_sendee">#ececec</color>
```
Comment (eddieberklee) [inline app/src/main/res/values/colors.xml]: Standardize on all capitals for the color codes please

File: app/src/main/java/com/compscieddy/meetinthemiddle/util/Util.java (hunk 6 above)
Comment (ambardhesi) [inline app/src/main/java/com/compscieddy/meetinthemiddle/util/Util.java]: Yeah my bad, this was an older method I was using

File: app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java [hunk 11]
```diff
@@ -1,68 +1,72 @@
...
+  TextView messageText;
+  Context context;
+
 
   public ChatHolder(View itemView) {
     super(itemView);
     rootView = itemView;
+    context = rootView.getContext();
     ButterKnife.bind(ChatHolder.this, rootView);
+    messageText = (TextView) rootView.findViewById(R.id.message_text);
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java:38]: you can use ButterKnife here :)

File: app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java [hunk 12]
```diff
@@ -1,68 +1,72 @@
...
 
   public void setIsSender(boolean isSender) {
-    int color;
+    int bg_color;
+    int text_color;
     if (isSender) {
-      color = ContextCompat.getColor(rootView.getContext(), R.color.group_chat_background_color);
-      leftArrow.setVisibility(View.GONE);
-      rightArrow.setVisibility(View.VISIBLE);
+      bg_color = ContextCompat.getColor(context, R.color.chat_sender);
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java:45]: I would factor out the getColor into a variable like int chatSenderColor
